front/lib/api/assistant/configuration.ts

```diff
@@ -1030,3 +1030,148 @@ async function _createAgentDataSourcesConfigData(
     );
   return agentDataSourcesConfigRows;
 }
+
+export async function duplicateAgentConfiguration(
+  auth: Authenticator,
+  agentConfigurationId: string
```

Comment:
why not take an agentConfigurationType? you will be able to recreate everything from it and this will force the caller to give you a valid one.

In the route you can simply call getAgentConfiguration() and have a proper error if not found?